pkg/apis/network/v1/egressrouter_types.go

```diff
@@ -0,0 +1,273 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EgressRouter is a feature allowing the user to define an egress router
+// that acts as a bridge between pods and external systems. The egress router runs
+// a service that redirects egress traffic originating from a pod or a group of
+// pods to a remote external system or multiple destinations as per configuration.
+//
+//  More specifically, given an EgressRouter CR with <name>, the CNO will create and manage:
+// - A service called <name>
+// - An egress pod called <name>
+// - A NAD called <name>
+//
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// EgressRouter is a single egressrouter pod configuration object.
+// +k8s:openapi-gen=true
+// +kubebuilder:resource:path=egressrouters,scope=Namespaced
+// +kubebuilder:printcolumn:name="Condition",type=string,JSONPath=".status.conditions[*].type"
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=".status.conditions[*].status"
+type EgressRouter struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinProperties:=4
```

Comment:
do you need the `MaxProperties` and `MinProperties`?

Reply:
yea else users might give 
- redirect and http configs and we won't be able to control that. OR
- they might not provide any mode config and we won't be able to catch that.